### PR TITLE
[SPARK-45458][SQL] Convert IllegalArgumentException to SparkIllegalArgumentException in bitwiseExpressions

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to convert IllegalArgumentException to SparkIllegalArgumentException in bitwiseExpressions.

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Update existed UT.
- Pass GA

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk @cloud-fan

File: common/utils/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -92,6 +92,11 @@
     ],
     "sqlState" : "22003"
   },
+  "BIT_POSITION_OUT_OF_RANGE" : {
```
Comment (cloud-fan) [inline common/utils/src/main/resources/error/error-classes.json:95]: IIRC we have an error class for invalid function arguments, shall we reuse it?

Comment (MaxGekk) [reply common/utils/src/main/resources/error/error-classes.json:95]: We have the error class `INVALID_PARAMETER_VALUE`, @panbingkun could you re-use or create a sub-class there, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: common/utils/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (panbingkun) [reply common/utils/src/main/resources/error/error-classes.json:95]: Okay, let me to do it.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun and @cloud-fan for review.

**Pull request closed without merging**